package/CodePoint.roc

```diff
@@ -338,6 +339,34 @@ parsePartialUtf8 = \bytes ->
     else
         Err InvalidUtf8
 
+
+toStr : List CP -> Result Str [BadUtf8]
+toStr = \cps ->
+
+    # allocated extra space for the extra bytes as some CPs expand into 
+    # multiple U8s, so this minimises extra allocations
+    capacity = List.withCapacity (50 + List.len cps)
```

Comment:
No need to change, but might be worth scaling the extra bytes based on length of list? Like maybe `(List.len cps // 10) + List.len cps` or something.